File: app/src/main/java/com/example/personalorganiser/data/Current.java
package com.example.personalorganiser.data;

import com.google.gson.annotations.SerializedName;

public class Current {
    @SerializedName("last_updated_epoch")
    private long lastUpdatedEpoch;
    @SerializedName("last_updated")
    private String lastUpdated;
    @SerializedName("temp_c")
    private double tempC;
    @SerializedName("temp_f")
    private double tempF;
    @SerializedName("is_day")
    private int isDay;
    private Condition condition;
    @SerializedName("wind_mph")
    private double windMph;
    @SerializedName("wind_kph")
    private double windKph;
    @SerializedName("precip_mm")
    private double precipMm;
    private int humidity;
    private int cloud;
    @SerializedName("feelslike_c")
    private double feelslikeC;
    @SerializedName("feelslike_f")
    private double feelslikeF;
    private double uv;

    public double getTempC() { return tempC; }
    public Condition getCondition() { return condition; }
    public int getHumidity() { return humidity; }
    // Getters for all fields
    // Setters for all fields
}

File: app/src/main/java/com/example/personalorganiser/data/Current.kt
package com.example.personalorganiser.data

import com.google.gson.annotations.SerializedName

data class Current(
    @SerializedName("last_updated_epoch")
    val lastUpdatedEpoch: Long?,
    @SerializedName("last_updated")
    val lastUpdated: String?,
    @SerializedName("temp_c")
    val tempC: Double?,
    @SerializedName("temp_f")
    val tempF: Double?,
    @SerializedName("is_day")
    val isDay: Int?,
    val condition: Condition?,
    @SerializedName("wind_mph")
    val windMph: Double?,
    @SerializedName("wind_kph")
    val windKph: Double?,
    @SerializedName("precip_mm")
    val precipMm: Double?,
    val humidity: Int?,
    val cloud: Int?,
    @SerializedName("feelslike_c")
    val feelslikeC: Double?,
    @SerializedName("feelslike_f")
    val feelslikeF: Double?,
    val uv: Double?
)

data class Current(
    val tempC: Double?,
    val tempF: Double?,
    val humidity: Int?,
    val condition: Condition?
)